packages/cadl-ranch-specs/http/azure/core/basic/mockapi.ts

```diff
@@ -18,16 +18,6 @@ Scenarios.Azure_Core_Basic_createOrUpdate = passOnSuccess({
     body: { name: "Madge" },
   },
   response: { status: 200, body: json(validUser) },
-  handler: (req: MockRequest) => {
-    if (req.params.id !== "1") {
-      throw new ValidationError("Expected path param id=1", "1", req.params.id);
-    }
```

Comment:
Does `request.params.id = 1` in request works in validation for path parameter?

I got error `{"message":"Expected query param id=1 but got undefined","expected":"1"}` when running the test (on all the cases in this module that checks `id=1` in path parameter.

Reply:
This was due to the issue in the generated handler (`packages/cadl-ranch/src/app/app.ts`). Now, I have added a fix to the code and confirmed that it is working. Please refer the snapshot below:

![image](https://github.com/user-attachments/assets/4c4d9487-bd25-4940-ace5-89ebd345ba81)

I have updated this PR with the required code.

Note to self: Take that code change to `typespec` repository